contracts/src/core/interfaces/IEigenDADisperserRegistryV2.sol

```diff
@@ -0,0 +1,145 @@
+
+    /// @notice This event is emitted when a disperser is added to the default dispersers set.
+    /// @param disperserId The ID of the disperser that has been added to the default set.
+    event DefaultDisperserAdded(uint32 indexed disperserId);
+
+    /// @notice This event is emitted when a disperser is added to the on-demand dispersers set.
+    /// @param disperserId The ID of the disperser that has been added to the on-demand set.
+    event OnDemandDisperserAdded(uint32 indexed disperserId);
+
+    /// @notice This event is emitted when a disperser is removed from the default dispersers set.
+    /// @param disperserId The ID of the disperser that has been removed from the default set.
+    event DefaultDisperserRemoved(uint32 indexed disperserId);
+
+    /// @notice This event is emitted when a disperser is removed from the on-demand dispersers set.
+    /// @param disperserId The ID of the disperser that has been removed from the on-demand set.
+    event OnDemandDisperserRemoved(uint32 indexed disperserId);
+
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    /// @notice This error is thrown when a zero address is provided as input where a valid address is required.
+    error InputAddressZero();
+
+    /// @notice This error is thrown when attempting to register a disperser that has already been registered.
+    error AlreadyRegistered();
+
+    /// @notice This error is thrown when an invalid signature is provided for verification.
+    error InvalidSignature();
+
+    /// @notice This error is thrown when attempting an operation on a disperser that is not registered.
+    error NotRegistered();
+
+    /// @notice This error is thrown when attempting to add a disperser that already exists in the specified set.
+    error DisperserAlreadyExists();
+
+    /// @notice This error is thrown when attempting to remove a disperser that does not exist in the specified set.
+    error DisperserNotFound();
+
+    /// @notice This error is thrown when attempting to perform an operation on a disperser that is in the default or on-demand dispersers sets.
+    error DisperserInSet();
+
+    /// -----------------------------------------------------------------------
+    /// External Functions
+    /// -----------------------------------------------------------------------
+
+    /// @notice This function registers a new disperser with the registry and assigns it a unique ID.
+    /// @param disperser The address of the disperser that should be registered.
+    /// @param relayURL The relay URL that will be associated with this disperser.
+    /// @return disperserId The unique ID that has been assigned to the newly registered disperser.
+    function registerDisperser(address disperser, string memory relayURL) external returns (uint32 disperserId);
```

Comment:
are we moving to single relayURL per disperser now? Up to now our disperser was associated to many relays which each had their own URL